### PR TITLE
feat(dashboard,medusa): Draft order detail

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Adds draft order details page
- Adds Shipping and Billing address forms to both draft order and order pages
- Adds Email form to both draft order and order pages
- Adds transfer ownership form to draft order, order and customer pages
- Update Combobox component allowing it to work with async data (`useInfiniteQuery`)

**@medusajs/medusa**
- Include country as a default relation of draft order addresses

### CODE REVIEW

File: packages/admin-next/dashboard/src/components/common/combobox/combobox.tsx [hunk 1]
```diff
@@ -12,217 +12,318 @@ import * as Popover from "@radix-ui/react-popover"
...
+    }
+
+    if (!isValueControlled) {
+      setUncontrolledValue(newValues || emptyState)
+    }
+    if (onChange) {
+      onChange(newValues)
+    }
+
+    setUncontrolledSearchValue("")
```
Comment (riqwan) [inline packages/admin-next/dashboard/src/components/common/combobox/combobox.tsx:98]: shouldn't this be guarded against `isArrayValue`? On line 88 its being set to the label and here we're setting it back to ""

**Comment by olivermrbl** [comment]
UI-wise LGTM!

Not sure if the following error is directly related to this PR or something more general, but upon opening "Transfer Ownership", the following error is thrown client-side:

CleanShot 2024-03-15 at 08 36 50

**Review by olivermrbl — APPROVED** [review]

**Comment by kasperkristensen** [comment]
@olivermrbl The reason the transfer orders modal doesn't work is because we are building the dashboard inside of our monorepo with local versions of our internal dependencies. 

This means that yarn will symlink `medusa-react` into `dashboard` including its node_modules. We then end up with two copies of `@tanstack/react-query` in `/dashboard/node_modules` and that causes this error. In the modal I am using `useInfiniteQuery` that gets imported from `node_modules/@tanstack/react-query` where every other query is importing it from `/node_modules/medusa-react/node_modules/@tanstack/...`. It won't occur in "real-life", and to get around it we can install `medusa-react@latest` instead of `workspace:^`, but that is currently missing a couple of new hooks, so will have to be after the next release. You can test it out in local now though, as it's not an issue in development, only production 👍

**Review by riqwan — APPROVED** [review]
lgtm!

**Pull request merged**